spec/beaker/host_spec.rb

```diff
@@ -487,6 +487,38 @@ module Beaker
           host.do_scp_to *args
         end
       end
+      context "using an ignore array with an absolute source path in host root" do
+        let( :source_path ) { '/puppetlabs-inifile' }
+        let( :target_path ) { '/etc/puppetlabs/modules/inifile' }
+
+        before :each do
+          test_dir = "#{source_path}/tests"
+          other_test_dir = "#{source_path}/tests/tests2"
+          another_test_dir = "#{source_path}/tests/tests3"
+
+          files = [
+              '00_EnvSetup.rb', '035_StopFirewall.rb', '05_HieraSetup.rb',
+              '01_TestSetup.rb', '03_PuppetMasterSanity.rb',
+              '06_InstallModules.rb','02_PuppetUserAndGroup.rb',
+              '04_ValidateSignCert.rb', '07_InstallCACerts.rb'              ]
+
+          @fileset1 = files.shuffle.map {|file| test_dir + '/' + file }
+          @fileset2 = files.shuffle.map {|file| other_test_dir + '/' + file }
+          @fileset3 = files.shuffle.map {|file| another_test_dir + '/' + file }
+
+          create_files( @fileset1 )
+          create_files( @fileset2 )
+          create_files( @fileset3 )
+        end
+
+        it "should create target dirs with correct path seperator" do
+          created_target_path = File.join(target_path, File.basename(source_path))
+          expect( host ).to receive( :mkdir_p ).with("#{created_target_path}/tests")
+          expect( host ).to receive( :mkdir_p ).with("#{created_target_path}/tests/tests2")
+          expect( host ).to receive( :mkdir_p ).with("#{created_target_path}/tests/tests3")
```

Comment:
@derdanne I believe you are missing the call 'host.do_scp_to *args' for this test.